curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -93,76 +144,174 @@ You should define a function `removeLastLunch`.
 
 ```js
-lunches = [];
-assert.strictEqual(removeFirstLunch(), "No lunches to remove.");
+const temp = console.log;
+console.log = () => {}
+try {
+  assert.deepStrictEqual(removeLastLunch(["Sushi", "Pizza", "Noodles"]), ["Sushi", "Pizza"]);
+  assert.deepStrictEqual(removeLastLunch(["Pizza", "Burger"]), ["Pizza"]);
+} finally {
+  console.log = temp;
+}
 ```
 
-When `lunches = ["Salad", "Eggs", "Cheese"]`, the function `removeFirstLunch` should return the string `"Salad removed from the start of the lunch menu."`.
+You should define a function `removeFirstLunch`.
 
 ```js
-lunches = ["Salad", "Eggs", "Cheese"];
-assert.strictEqual(removeFirstLunch(), "Salad removed from the start of the lunch menu.");
+assert.isFunction(removeFirstLunch);
+```
+
+The function `removeFirstLunch` should have a single parameter.
 
-// prevent hardcoding
-assert.strictEqual(removeFirstLunch(), "Eggs removed from the start of the lunch menu.");
-assert.strictEqual(removeFirstLunch(), "Cheese removed from the start of the lunch menu.");
+```js
+assert.lengthOf(removeFirstLunch, 1);
 ```
 
-You should define a function `getRandomLunch`.
+When the input array is empty, the function `removeFirstLunch` should log the string `"No lunches to remove."` to the console.
 
 ```js
-assert.isFunction(getRandomLunch);
+const testLunches = [];
+const tempArr = [];
+const temp = console.log;
+try {
+  console.log = obj => tempArr.push(obj);
+  removeFirstLunch(testLunches);
+  assert.strictEqual(tempArr[0], "No lunches to remove.");
+} finally {
+  console.log = temp;
+}
 ```
 
-When the `lunches` array is empty, the function `getRandomLunch` should return the string `"No lunches available."`.
+`removeFirstLunch(["Salad", "Eggs", "Cheese"])` should log the string `"Salad removed from the start of the lunch menu."` to the console.
 
 ```js
-lunches = [];
-assert.strictEqual(getRandomLunch(), "No lunches available.");
+let testLunches = ["Salad", "Eggs", "Cheese"];
+const tempArr = [];
+const temp = console.log;
+try {
+  console.log = obj => tempArr.push(obj);
+  removeFirstLunch(testLunches);
+  assert.strictEqual(tempArr[0], "Salad removed from the start of the lunch menu.");
+  testLunches = ["Stew", "Soup", "Toast"]
+  removeFirstLunch(testLunches);
+  assert.strictEqual(tempArr[1], "Stew removed from the start of the lunch menu.");
+} finally {
+  console.log = temp;
+}
 ```
 
-When `lunches = ["Fish", "Fries", "Roast"]`, the function `getRandomLunch` should return a string in the format `"Randomly selected lunch: [Lunch Item]"`.
+`removeFirstLunch(["Sushi", "Pizza", "Burger"])` should return `["Pizza", "Burger"]`.
 
 ```js
-const temp = Math.random;
+const temp = console.log;
+console.log = () => {}
+try {
+  assert.deepStrictEqual(removeFirstLunch(["Sushi", "Pizza", "Burger"]), ["Pizza", "Burger"]);
+  assert.deepStrictEqual(removeFirstLunch(["Pizza", "Burger"]), ["Burger"]);
+} finally {
+  console.log = temp;
+}
+```
 
```

Comment:
One last thing:
````suggestion

`addLunchToEnd`, `addLunchToStart`, `removeLastLunch`, and `removeFirstLunch` should return the same array passed as an argument after updating it.

```js
const temp = console.log;
console.log = () => {}
const testLunches = ["Spam"];
try {
  assert.equal(addLunchToEnd(testLunches, "Pizza"), testLunches);
  assert.equal(addLunchToStart(testLunches, "Caviar"), testLunches);
  assert.equal(removeLastLunch(testLunches), testLunches);
  assert.equal(removeFirstLunch(testLunches), testLunches);
} finally {
  console.log = temp;
}
```

````
We should add this just in case someone returns a copy of the array instead of the original.